NumLib/IndexValueVector.h

```diff
@@ -0,0 +1,27 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUM_LIB_INDEXVALUEVECTOR_H
+#define NUM_LIB_INDEXVALUEVECTOR_H
+
+#include <vector>
+
+namespace NumLib
+{
+
+template <typename IndexType>
+struct IndexValueVector final
```

Comment:
I think this data type can also be used for Neumann or other conditions. In that case, the values are not known solutions any more.